pulp_rpm/app/models/package.py

```diff
@@ -354,4 +354,10 @@ def list_to_createrepo_c(l):
         package.url = getattr(self, PULP_PACKAGE_ATTRS.URL)
         package.version = getattr(self, PULP_PACKAGE_ATTRS.VERSION)
 
+        if checksum_type:
+            pkgId = getattr(self.contentartifact_set.first().artifact, checksum_type.lower(), None)
+            if pkgId:
+                package.checksum_type = checksum_type.lower()
```

Comment:
there is no need to .lower() because on the imput uppercase would pass the validation
```

{
    "metadata_checksum_type": [
        "\"SHA512\" is not a valid choice."
    ]
}
```